### PR TITLE
Oauth interop

### PR DESCRIPTION
**Pull request by yang-g**
This is based on #2385 which should be merged first.

-- Split the part in fetch_oauth2.c of getting oauth2 access token from a credentials object
-- Use the previous part to create a channel with AccessTokenCredentials
-- Add an interop test case according to https://github.com/grpc/grpc/blob/master/doc/interop-test-descriptions.md#oauth2_auth_token

Tested manually for both grpc_fetch_oauth2 and interop_client

### CODE REVIEW

File: test/core/security/oauth2_utils.c [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+#include <grpc/support/slice.h>
+#include <grpc/support/sync.h>
+
+#include "src/core/security/credentials.h"
+
+typedef struct {
+  grpc_pollset pollset;
+  int is_done;
+  char *token;
+} synchronizer;
```
Comment (jboeuf) [inline test/core/security/oauth2_utils.c]: Maybe it's not that much of a synchronizer anymore... How about oauth2_request instead.

File: test/cpp/interop/client_helper.cc [hunk 2]
```diff
@@ -73,6 +86,20 @@ grpc::string GetServiceAccountJsonKey() {
...
+grpc::string GetOauth2AccessToken() {
+  std::shared_ptr<Credentials> creds = CreateServiceAccountCredentials();
+  SecureCredentials* secure_creds =
+      dynamic_cast<SecureCredentials*>(creds.get());
+  GPR_ASSERT(secure_creds != nullptr);
+  grpc_credentials* c_creds = secure_creds->GetRawCreds();
+  char* token = grpc_test_fetch_oauth2_token_with_credentials(c_creds);
+  GPR_ASSERT(token != nullptr);
+  gpr_log(GPR_INFO, "Get raw oauth2 access token: %s", token);
+  grpc::string access_token(token + sizeof("Bearer ") - 1);
```
Comment (jboeuf) [inline test/cpp/interop/client_helper.cc]: why `-1`? I'm sure it works but I don't understand why :).

**Comment by jboeuf** [comment]
A few small nits. LGTM otherwise.

File: test/core/security/oauth2_utils.c (hunk 1 above)
Comment (yang-g) [inline test/core/security/oauth2_utils.c]: Done.

File: test/cpp/interop/client_helper.cc (hunk 2 above)
Comment (yang-g) [inline test/cpp/interop/client_helper.cc]: sizeof seems to include the terminating null.

Comment (jboeuf) [inline test/cpp/interop/client_helper.cc]: ooops. That's right. thanks!

**Comment by jboeuf** [comment]
I'm happy to merge but travis and jenkins are red. Looks unrelated though. Can you confirm?

**Comment by jboeuf** [comment]
@yang-g Are you OK with me merging this despite the test failures?

**Comment by yang-g** [comment]
I will first update the PR according to a spec change in #2422 and then push on #2385. Will ping you after those are done. Sorry for the delay.

**Comment by yang-g** [comment]
Updated. Seems green enough. PTAL.

**Pull request merged**